### PR TITLE
[FR] End of chapter 3 + chapter 4 

### PR DESCRIPTION
**Pull request by lbourdois**
Hi,

I took the liberty of adding the end of chapter 3 (the beginning being #131) as well as chapter 4 to complete a first version of the course in French. 

A second version that will be able to start will consist in reviewing all the chapters in order to propose a homogeneous translation.

I also made some minor corrections on English files.

### CODE REVIEW

**Comment by lbourdois** [comment]
@lewtun I'm going to need your help because the error message "Error: The operation was cancelled." is not very informative for me to try to solve the check myself.

**Comment by lewtun** [comment]
> @lewtun I'm going to need your help because the error message "Error: The operation was cancelled." is not very informative for me to try to solve the check myself.

Yeah the CI gets cancelled if you push a commit while it's running. Most of the time it restarts correctly, but sometimes it needs a manual restart - I've just done that :)

File: chapters/en/chapter3/3.mdx [hunk 1]
```diff
@@ -162,7 +162,7 @@ This time, it will report the validation loss and metrics at the end of each epo
 
 The `Trainer` will work out of the box on multiple GPUs or TPUs and provides lots of options, like mixed-precision training (use `fp16 = True` in your training arguments). We will go over everything it supports in Chapter 10.
 
-This concludes the introduction to fine-tuning using the `Trainer` API. An example of doing this for most common NLP tasks will be given in Chapter 7, but for now let's look at how to do the same thing in pure PyTorch.
+This concludes the introduction to fine-tuning using the `Trainer` API. An example of doing this for most common NLP tasks will be given in [Chapter 7](course/chapter7), but for now let's look at how to do the same thing in pure PyTorch.
```
Comment (lewtun) [inline chapters/en/chapter3/3.mdx:165]: Nice catch!

File: chapters/en/chapter3/6.mdx [hunk 2]
```diff
@@ -251,7 +251,7 @@ Test what you learned in this chapter!
 			explain: "Almost! There are some small additional changes required. For example, you need to run everything in a <code>TPUStrategy</code> scope, including the initialization of the model."
 		},
 		{
-			text: "You can leverage existing methods such as <code>compile()</code>, <code>fit()<c/ode>, and <code>predict()</code>.",
+			text: "You can leverage existing methods such as <code>compile()</code>, <code>fit()</code>, and <code>predict()</code>.",
```
Comment (lewtun) [inline chapters/en/chapter3/6.mdx:254]: Thanks for the fix!

**Review by lewtun — APPROVED** [review]
Thanks for translating these final sections of the French course @lbourdois 🤗 ! Also thank you for catching some bugs in the English version 🙏

**Comment by lewtun** [comment]
The build looks good, so I'll merge this and prepare a new deploy on hf.co - thanks!

File: chapters/en/chapter3/3.mdx (hunk 1 above)
Comment (lbourdois) [reply chapters/en/chapter3/3.mdx:165]: Yes, thank you very much. I'll make a note to correct this in my next PR.

**Pull request merged**